### ISSUE TITLE
EPUB page-list UI (for page-breaks)

### ISSUE THREAD

**Issue by danielweck** [readium/readium-desktop#82]
This is implemented in `r2-testapp-js`, but is currently missing in `readium-desktop`.

**Comment by danielweck** [comment]
Related: https://github.com/edrlab/readium-desktop/issues/81

**Comment by danielweck** [comment]
Related: https://github.com/edrlab/readium-desktop/issues/20

**Comment by danielweck** [comment]
Perhaps use a tab user interface? 
https://github.com/reactjs/react-tabs

**Comment by danielweck** [comment]
W3C `public-publ-wg` conversation thread here: https://lists.w3.org/Archives/Public/public-publ-wg/2018Mar/0072.html

**Comment by clebeaupin** [comment]
Check this issue on children literature epub

**Comment by danielweck** [comment]
I am not seeing the page list UX in the current Thorium. Re-opening. See screenshots:

![ThoriumPageList](https://user-images.githubusercontent.com/625201/63063264-df0cd000-bef3-11e9-894a-627be1e2a602.png)

This is `r2-testapp-js`:

![R2TestAppPageList](https://user-images.githubusercontent.com/625201/63063268-e2a05700-bef3-11e9-9d26-f402420404a7.png)

**Comment by danielweck** [comment]
Examples:
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/georgia-pls-ssml/EPUB/nav.xhtml#L46-L56
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/page-blanche/EPUB/Navigation/nav.xhtml#L22-L33
https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/haruko-html-jpeg/OPS/xhtml/toc.xhtml#L22-L67

**Comment by llemeurfr** [comment]
A page list is simply a large list of numbers. Displaying it as a looong vertical list is pretty boring.

Therefore we had planned the following initially with the designer: 
- if there the publication has an associated Page List ...
- display a "Go To Page" component in the "navigation" panel, followed  by an input field

![Capture d'écran 2019-08-29 15 22 53](https://user-images.githubusercontent.com/14943186/63943936-e7771600-ca70-11e9-9b02-de8f6faf29c6.png)

- let the user enter a string (it may be "IX", so not a number) 
- looks for the same string in the page list and follows the link. Trim leading 0's and spaces before search. 

At the same time, please hide "Annotations" (not implemented yet).

**Comment by danielweck** [comment]
A word of caution: the EPUB3 NavDoc "page list" (i.e. `nav` `epub:type="page-list"` and/or `role="doc-pagelist"`) and the associated HTML Content Document's `epub:type="pagebreak"` (and/or `role="doc-pagebreak"`) elements (with `title` or `aria-label` attribute, or actual visible child textual content) do not necessarily specify integers ... they can be roman numerals too.

Example with Adobe Digital Editions:

<img width="170" alt="ADE_PageList" src="https://user-images.githubusercontent.com/625201/63964009-12239780-ca8e-11e9-8bc1-6614e8c8e82e.png">

That being said, of course I agree that allocating a long scrolling list in the GUI seems like a waste of screen real estate. I think a pull-down list (combo box / select control) would work better, also with the "goto page" text input field.

Specification reference:
https://www.w3.org/publishing/epub32/epub-packages.html#sec-nav-pagelist
... and example:
https://www.w3.org/publishing/epub32/epub-packages.html#example-40

**Comment by danielweck** [comment]
Just to clarify my comment: this means I object to **not** exposing the "page list" at all, because users cannot possibly know what to type in the "goto page" text input field (which is actually a search criteria for arbitrary data, e.g. roman numerals, integers, and possibly other syntaxes).

**Comment by JayPanoz** [comment]
re i18n, not sure about that but the W3C layout requirements mention [Abjad page numbering](https://www.w3.org/TR/alreq/#h_pages), which is tough to Google actually ([best result might be this Wikipedia entry](https://en.wikipedia.org/wiki/Abjad_numerals)) but I guess this would automatically be covered by the roman numerals’ case.

**Comment by danielweck** [comment]
Interesting info master Jiminy, thanks :)

**Comment by llemeurfr** [comment]
Daniel, Acrobat Reader offers a "go to" only and sees massive deployment. 

The main use case, as far as I know, is the famous teacher requesting students to go to a given page. No discovery there. Adding a list would be an added work, a loss of space, an ugly UI and a burden for React (that we can only solve by list virtualization). 

Proposal: for v1.0, implement go to page only. Check comments. If people complain find a smart way to offer discovery.

**Comment by danielweck** [comment]
Laurent - there is flexibility in the interpretation of the a11y success criteria, so I agree with your proposal to *not* render the pagelist (i.e. EPUB2 NCX `pageList/pageTarget/navLabel` and EPUB3 NavDoc `nav@epub:type="pagelist"`).

EpubTest.org:
"Fundamental Accessibility Tests: Basic Functionality"
http://epubtest.org/testbooks

* Check if you can navigate the content by pages
* Use the "Next/Previous Page" or "Go to Page" or the "Page List" feature in the Reading system and check if you are able to navigate to different pages. Check that you are able to determine which page you are currently in.
* This EPUB file contains 10 hard coded page breaks. Note that these page breaks are different from the reflowed page numbers.

GitHub source:
https://github.com/daisy/epub-accessibility-tests/blob/f04c9d6dd1c4c9275e9f80000b31663fb06b44b2/content/epub30-test-0301/EPUB/xhtml/nav.xhtml#L52-L66
...and:
https://github.com/daisy/epub-accessibility-tests/blob/f04c9d6dd1c4c9275e9f80000b31663fb06b44b2/content/epub30-test-0301/EPUB/xhtml/Basic-functionality-tests.xhtml#L86-L97

**Comment by Scarsniik** [comment]
@danielweck What is the function or the list of links that I need to go to a specific page ?

**Comment by danielweck** [comment]
@Scarsniik as discussed in Slack audio chat:

https://github.com/readium/r2-shared-js/blob/61f9d496886d9dfe0420c9b445e6b09ea21499f7/src/models/publication.ts#L81-L85

https://github.com/readium/r2-shared-js/blob/61f9d496886d9dfe0420c9b445e6b09ea21499f7/src/parser/epub.ts#L151

Test with EPUBs (for example):

* https://idpf.github.io/epub3-samples/30/samples.html#childrens-literature ( https://github.com/IDPF/epub3-samples/blob/0b3e80553f509fbdfe591008cd7d1e804b24db54/30/childrens-literature/EPUB/nav.xhtml#L146-L426 )
* https://github.com/daisy/epub-accessibility-tests/blob/f04c9d6dd1c4c9275e9f80000b31663fb06b44b2/content/epub30-test-0301/EPUB/xhtml/nav.xhtml#L52-L66
* https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/georgia-pls-ssml/EPUB/nav.xhtml#L46-L56
* https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/page-blanche/EPUB/Navigation/nav.xhtml#L22-L33
* https://github.com/IDPF/epub3-samples/blob/68ba020fe5c99625d45dd50a94c4b202c5f220e1/30/haruko-html-jpeg/OPS/xhtml/toc.xhtml#L22-L67

**Comment by danielweck** [comment]
Fixed via https://github.com/edrlab/thorium-reader/pull/1319
